app/controllers/posts_controller.rb

```diff
@@ -1,39 +1,28 @@
 class PostsController < ApplicationController
   skip_before_action :verify_authenticity_token
-  load_and_authorize_resource
   def index
-    @user = User.includes(:posts).find(params[:user_id])
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
   end
 
   def show
-    @user = User.find(params[:user_id])
-    @post = @user.posts.includes(:comments, :likes).find(params[:id])
+    @post = Post.find(params[:id])
   end
 
   def new
```

Comment:
User is not able to create a post. Kindly make sure that a signed in user can create a post.

Reply:
Fixed